bootcompiler/src/main/scala/org/mozartoz/bootcompiler/transform/OnStackMarking.scala

```diff
@@ -30,7 +30,7 @@ sealed class VarState(val onStack: Boolean, val locked: Boolean) {
   }
 }
 object Declared      extends VarState(true, false)
-object Captured      extends VarState(false, false)
+object Captured      extends VarState(false, true) // TODO false false if frame hierarchy, false true if extraction
```

Comment:
You could read the option in Scala as well to make this correct based on the option value.
It's not ideal but pragmatic for now and that's what I did for self-tail calls in 45edd9a0e3467e9e7bc20409b3cc82097a660f58.